### PR TITLE
Added Firmware attestation feature for Enterprise and Opal drives

### PR DESCRIPTION
**Pull request by adithya-22**
This feature is supported on Enterprise & Opal drives and supports the following:

1. Pulls Tper attestation certificate out from the drive
2. Performs firmware attestation
3. Receives attestation response, parses output, verifies original string and signature, prints measurement data

### CODE REVIEW

File: TCGstorageAPI/tcgapi_test.py [hunk 1]
```diff
@@ -535,6 +538,51 @@ def test_tperSign_cert_fail(self):
         
         self.sedmock.invoke.return_value = status, rv, kwrv = (0x01, None, None)
         self.assertFalse(self.sed.get_tperSign_cert())
+    
+    def test_tper_attestation_cert_success(self):
+
+        self.sedmock.invoke.return_value = status, rv, kwrv = (0, [self.sed.cert], {})
+        type(self.sedmock).SSC = mock.PropertyMock(return_value='Enterprise')
```
Comment (simonpha) [inline TCGstorageAPI/tcgapi_test.py]: @adithya-22 Question, do we just test the value for Enterprise for the mock but not Opal?

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 boost
```
Comment (simonpha) [inline requirements.txt:1]: @adithya-22 If we are specifying version numbers for cryptography and pyopenssl can we do it for passlib and pycparser as well?

Comment (adithya-22) [reply requirements.txt:1]: okay, sure.

File: samples/sample_cli.py [hunk 3]
```diff
@@ -187,6 +187,48 @@ def TlsOperation(self, args=None):
...
+        # Retrieve the Tper attestation certificate
+        att_cert = self.sed.get_tperAttestation_Cert()
+        if (len(att_cert)) == 0:
+            print("The drive does not contain a certificate")
+            return   
+            
+        # Validate the drive attestation certificate against the root certificate
+        identity = verifyidentity.VerifyIdentity(att_cert)
+        identity.validate_drive_cert()
+        acc_nonce,sub_name='23helloseagate',identity.CN,
```
Comment (simonpha) [inline samples/sample_cli.py]: Can we add a comment here for what the acc_nounce, sub_name, and acc_ID is here? Then we should say that these values are simulated for the sample script.

File: TCGstorageAPI/tcgapi_test.py (hunk 1 above)
Comment (adithya-22) [reply TCGstorageAPI/tcgapi_test.py]: I think I might have missed that. Will add another test case for Opal.

File: samples/sample_cli.py (hunk 3 above)
Comment (adithya-22) [reply samples/sample_cli.py]: okay, sure.

**Review by simonpha — APPROVED** [review]

**Review by Keerthana-Bidarakoppa — APPROVED** [review]

**Pull request merged**